google/cloud/bigtable/tools/run_emulator_utils.sh

```diff
@@ -35,60 +35,97 @@ function kill_emulators {
   echo " done."
 }
 
-function start_emulators {
-  echo "Launching Cloud Bigtable emulators in the background"
-  trap kill_emulators EXIT
+################################################
+# Wait until the port number is printed in the log file for one of the
+# emulators
+# Globals:
+#   None
+# Arguments:
+#   logfile: the name of the logfile to wait on.
+# Returns:
+#   None
+################################################
+wait_for_port() {
+  local -r logfile="$1"
+  shift
 
-  # The tests typically run in a Docker container, where the ports are largely
-  # free; when using in manual tests, you can set EMULATOR_PORT.
-  local -r emulator_port=${EMULATOR_PORT:-9000}
-  "${CBT_EMULATOR_CMD}" -port "${emulator_port}" >emulator.log 2>&1 </dev/null &
-  EMULATOR_PID=$!
+  local emulator_port="0"
+  local -r expected='Cloud Bigtable emulator running on'
+  for attempt in $(seq 1 8); do
+    if head -1 "${logfile}" | grep -q "${expected}"; then
```

Comment:
It is worth baking in knowledge here that "running on" is the first line in the logfile?  That is, just grep, don't head.

Reply:
Fixed.